pkg/minikube/cluster/cluster.go

```diff
@@ -229,11 +230,38 @@ var assets = []fileToCopy{
 	},
 }
 
-func updateLocalkubeFromURL(config KubernetesConfig, client *ssh.Client) error {
+func (k *KubernetesConfig) getLocalkubeCacheFilepath() string {
+	return filepath.Join(constants.Minipath, "cache", "localkube",
+		filepath.Base(url.QueryEscape("localkube-"+k.KubernetesVersion)))
+}
+
+func (k *KubernetesConfig) isLocalkubeCached() bool {
```

Comment:
I think these might make more sense on a new struct, instead of on the KubernetesConfig. Maybe something called:

localKubeCacher.

It could have a KubernetesConfig (or just the version parameter as a member. cluster.go is getting pretty large, so It should probably go in a new file as well (you can keep it part of the same package though).